src/ck-core/ckrdmadevice.C

```diff
@@ -178,13 +165,14 @@ int CkRdmaGetDestPEChare(int dest_pe, void* obj_ptr) {
 */
 
 static int findFreeIpcEvent(DeviceManager* dm, const size_t comm_offset) {
-  int pool_size = CsvAccess(gpu_manager).cuda_ipc_event_pool_size;
+  int pool_size = CsvAccess(gpu_manager).cuda_ipc_event_pool_size_pe;
+  int pool_start = CkMyRank() * pool_size;
   int device_index = dm->global_index;
   cuda_ipc_device_info& my_device_info = CsvAccess(gpu_manager).cuda_ipc_device_infos[device_index];
 
   // Free IPC events that are complete
   // TODO: Don't do this every time but only when the event pool is somewhat empty
-  for (int i = 0; i < pool_size; i++) {
+  for (int i = pool_start; i < pool_start + pool_size; i++) {
```

Comment:
This changes the CUDA IPC event pool to be exclusive to be PE, instead of the pool being shared by multiple PEs. This should reduce contention, but also limits the number of IPC events that can be used by a PE. This shouldn't be an issue unless the number of messages being sent by each PE are very disproportionate.